app/controllers/friendships_controller.rb

```diff
@@ -1,14 +1,17 @@
 class FriendshipsController < ApplicationController
   def create
-    @friendship = current_user.friendships.build(friend_id: params[:friend_id])
-    if @friendship.save
-      flash[:notice] = "Added friend"
+    if current_user
+      @friendship = current_user.friendships.build(friend_id: params[:friend_id])
+      if @friendship.save
+        flash[:notice] = "Added friend"
+      else
+        friend = @friendship.friend
+        flash[:notice] = "Error Adding Friend"
+      end
     else
-      friend = @friendship.friend
-      flash[:notice] = "Error Adding Friend"
+      flash[:notice] = "You're not signed in!"
```

Comment:
Functionality works just fine! I would just change the wording into something like 'Sign up or Log in to start adding friends' :)